src/autoscaler/models/health.go

```diff
@@ -1,5 +1,48 @@
 package models
 
+import (
+	"fmt"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
 type HealthConfig struct {
-	Port int `yaml:"port"`
+	Port                    int    `yaml:"port"`
+	HealthCheckUsername     string `yaml:"username"`
+	HealthCheckUsernameHash string `yaml:"username_hash"`
+	HealthCheckPassword     string `yaml:"password"`
+	HealthCheckPasswordHash string `yaml:"password_hash"`
+}
+
+func (c *HealthConfig) Validate(component string) error {
```

Comment:
I wonder if we need a component name here. The log of each component writes to its own file, so user won't get confused which component has error.

Reply:
@qibobo  I think that is a reasonable argument. I will do the necessary changes.